migration/sql-files/062-workitem-id-unique-per-space.sql

```diff
@@ -0,0 +1,91 @@
+ALTER TABLE work_item_links DROP CONSTRAINT work_item_links_source_id_fkey;
+ALTER TABLE work_item_links DROP CONSTRAINT work_item_links_target_id_fkey;
+ALTER TABLE work_item_revisions DROP CONSTRAINT work_item_revisions_work_items_fk;
+ALTER TABLE work_items DROP CONSTRAINT work_items_pkey;
+DROP TRIGGER upd_tsvector ON work_items;
+DROP FUNCTION IF EXISTS workitem_tsv_TRIGGER() CASCADE;
+DROP INDEX IF EXISTS fulltext_search_index;
+
+
+-- RENAME COLUMNs of other tables referencing a work item id
+ALTER TABLE work_item_links RENAME COLUMN "source_id" TO "source_id_old";
+ALTER TABLE work_item_links RENAME COLUMN "target_id" TO "target_id_old";
+ALTER TABLE work_item_link_revisions RENAME COLUMN "work_item_link_source_id" TO "work_item_link_source_id_old";
+ALTER TABLE work_item_link_revisions RENAME COLUMN "work_item_link_target_id" TO "work_item_link_target_id_old";
+ALTER TABLE work_item_revisions RENAME COLUMN "work_item_id" TO "work_item_id_old";
+
+-- ADD new COLUMNs
+ALTER TABLE work_item_links ADD COLUMN "source_id" UUID;
+ALTER TABLE work_item_links ADD COLUMN "target_id" UUID;
+ALTER TABLE work_item_link_revisions ADD COLUMN "work_item_link_source_id" UUID;
+ALTER TABLE work_item_link_revisions ADD COLUMN "work_item_link_target_id" UUID;
+ALTER TABLE work_item_revisions ADD COLUMN "work_item_id" UUID;
+
+-- assign new UUIDs TO the existing work items
+ALTER TABLE work_items DROP COLUMN "id";
+ALTER TABLE work_items ADD COLUMN "id" UUID default uuid_generate_v4();
+UPDATE work_items SET id = uuid_generate_v4();
+-- apply generated UUIDs in other tables
+UPDATE work_item_links SET source_id = w.id FROM work_items w WHERE w.number = work_item_links.source_id_old;
+UPDATE work_item_links SET target_id = w.id FROM work_items w WHERE w.number = work_item_links.target_id_old;
+UPDATE work_item_link_revisions SET work_item_link_source_id = w.id FROM work_items w WHERE w.number = work_item_link_revisions.work_item_link_source_id_old;
+UPDATE work_item_link_revisions SET work_item_link_target_id = w.id FROM work_items w WHERE w.number = work_item_link_revisions.work_item_link_target_id_old;
+UPDATE work_item_revisions SET work_item_id = w.id FROM work_items w WHERE w.number = work_item_revisions.work_item_id_old;
+UPDATE comments SET parent_id = w.id FROM work_items w WHERE w.number::text = comments.parent_id;
+
+-- DROP old COLUMNs
+ALTER TABLE work_item_links DROP COLUMN "source_id_old";
+ALTER TABLE work_item_links DROP COLUMN "target_id_old";
+ALTER TABLE work_item_revisions DROP COLUMN "work_item_id_old";
+ALTER TABLE work_item_link_revisions DROP COLUMN "work_item_link_source_id_old";
+ALTER TABLE work_item_link_revisions DROP COLUMN "work_item_link_target_id_old";
+
+-- recreate CONSTRAINTs and TRIGGERs
+ALTER TABLE work_items ADD CONSTRAINT work_items_pkey PRIMARY KEY (id);
+ALTER TABLE work_item_links ADD CONSTRAINT work_item_links_source_id_fkey FOREIGN KEY (source_id) REFERENCES work_items(id) ON DELETE CASCADE;
```

Comment:
I was just worried it is easy to forget adding a reference when doing it in two steps.